core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -154,15 +150,29 @@ class BillingProfileManagerDAOImpl(
 
     // create the profile
     val profileApi = apiClientProvider.getProfileApi(ctx)
-    val createProfileRequest = new CreateProfileRequest()
-      .tenantId(azureManagedAppCoordinates.tenantId)
-      .subscriptionId(azureManagedAppCoordinates.subscriptionId)
-      .managedResourceGroupId(azureManagedAppCoordinates.managedResourceGroupId)
-      .displayName(displayName)
-      .id(UUID.randomUUID())
-      .biller("direct") // community terra is always 'direct' (i.e., no reseller)
-      .cloudPlatform(CloudPlatform.AZURE)
-      .policies(policyInputs)
+    val createProfileRequest = billingInfo match {
+      case Left(billingAccountName) =>
+        val rawlsBillingAccountName = billingAccountName
+        new CreateProfileRequest()
+          .billingAccountId(rawlsBillingAccountName.withoutPrefix())
+          .displayName(displayName)
+          .id(UUID.randomUUID())
+          .biller("direct") // community terra is always 'direct' (i.e., no reseller)
+          .cloudPlatform(CloudPlatform.GCP)
+          .policies(policyInputs)
+
+      case Right(coords) =>
+        val azureManagedAppCoordinates = coords
+        new CreateProfileRequest()
+          .tenantId(azureManagedAppCoordinates.tenantId)
+          .subscriptionId(azureManagedAppCoordinates.subscriptionId)
+          .managedResourceGroupId(azureManagedAppCoordinates.managedResourceGroupId)
+          .displayName(displayName)
+          .id(UUID.randomUUID())
+          .biller("direct") // community terra is always 'direct' (i.e., no reseller)
+          .cloudPlatform(CloudPlatform.AZURE)
+          .policies(policyInputs)
+    }
```

Comment:
Could you break out the common set-up and only branch on the cloud-specific set-up?

(Sorry about my rusty Scala, the syntax may be off!)
```suggestion
    val createProfileRequest =
      new CreateProfileRequest()
          .displayName(displayName)
          .id(UUID.randomUUID())
          .biller("direct") // community terra is always 'direct' (i.e., no reseller)
          .policies(policyInputs)
    billingInfo match {
      case Left(billingAccountName) =>
        val rawlsBillingAccountName = billingAccountName
        createProfileRequest
          .billingAccountId(rawlsBillingAccountName.withoutPrefix())
          .cloudPlatform(CloudPlatform.GCP)
      case Right(coords) =>
        val azureManagedAppCoordinates = coords
        createProfileRequest
          .tenantId(azureManagedAppCoordinates.tenantId)
          .subscriptionId(azureManagedAppCoordinates.subscriptionId)
          .managedResourceGroupId(azureManagedAppCoordinates.managedResourceGroupId)
          .cloudPlatform(CloudPlatform.AZURE)
    }
```

Reply:
See my most recent commit.

Reply:
Looks good, thanks!